internal/resolve/catalog_test.go

```diff
@@ -935,3 +940,26 @@ func TestMultiplePriority(t *testing.T) {
 	assert.Nil(t, gotVersion)
 	assert.Nil(t, gotDeprecation)
 }
+
+func TestMultipleChannels(t *testing.T) {
+	pkgName := randPkg()
+	w := staticCatalogWalker{
+		"a": func() (*declcfg.DeclarativeConfig, *catalogd.ClusterCatalogSpec, error) {
+			return &declcfg.DeclarativeConfig{}, nil, nil
+		},
+		"b": func() (*declcfg.DeclarativeConfig, *catalogd.ClusterCatalogSpec, error) {
+			return &declcfg.DeclarativeConfig{}, nil, nil
+		},
+		"c": func() (*declcfg.DeclarativeConfig, *catalogd.ClusterCatalogSpec, error) {
+			return genPackage(pkgName), nil, nil
+		},
+	}
+	r := CatalogResolver{WalkCatalogsFunc: w.WalkCatalogs}
+	ce := buildFooClusterExtension(pkgName, "beta", "", ocv1alpha1.UpgradeConstraintPolicyEnforce)
+	ce.Spec.Source.Catalog.Channels = append(ce.Spec.Source.Catalog.Channels, "alpha")
```

Comment:
This seems like it could be annoying to maintain. Change `buildFooClusterExtension` to accept a slice of channels?